### DISCUSSION TITLE
Enable Kong Manager in read only mode

### DISCUSSION THREAD

**Discussion by elhackeado** [Kong/kong-manager#105] [Ideas]
Can we have a read only mode feature in Kong Manger where no edits are allowed via Kong Manager ? Since Kong decK is the recommended way of  managing state of the Kong Gateway, updating directly from Kong Manager can put Kong Gateway out of sync with decK configurations.

In my opinion, having Kong Manager in read only mode alongside kong decK can provide better visibility over the current configuration state of Kong Gateway, as one can quickly validate the configurations from Kong Manager rather having to read decK config files.

**Comment by sumimakito** [comment]
Hello @elhackeado - [DB-less mode](https://docs.konghq.com/gateway/latest/production/deployment-topologies/db-less-and-declarative-config/#main) might be a good fit for your requirement of read-only Admin APIs, and it should also allow you to export the declarative configuration with decK.
